com.ibm.streamsx.metrics/com.ibm.streamsx.metrics.service/MetricsIngestService.spl

```diff
@@ -0,0 +1,69 @@
+//
+// ****************************************************************************
+// * Copyright (C) 2017, International Business Machines Corporation          *
+// * All rights reserved.                                                     *
+// ****************************************************************************
+//
+
+/**
+ * A service which publishes the [com.ibm.streamsx.metrics::MetricsSource] 
+ * operator's tuples in JSON format to the 'streamsx/metrics/ingest' topic.
+ * 
+ * Output JSON Schema:
+ * {
+ *   "domainId" : string,
+ *   "instanceId" : string,
+ *   "jobId" : string,
+ *   "jobName" : string,
+ *   "resource" : string,
+ *   "peId" : long,
+ *   "operatorName" : string,
+ *   "channel" : int,
+ *   "portIndex" : int,
+ *   "connectionId" : string,
+ *   "metricType" : string,
+ *   "metricKind" : string,
+ *   "metricName" : string,
+ *   "metricValue" : long,
+ *   "lastTimeRetrieved" : long
+ * }
+ */
+ 
+namespace com.ibm.streamsx.metrics.service;
+
+use com.ibm.streamsx.metrics::* ;
+use com.ibm.streamsx.json::TupleToJSON ;
+use com.ibm.streamsx.topology.topic::Publish ;
+
+public composite MetricsIngestService
+{			
+	param
+		expression<rstring> $applicationConfigurationName ;
```

Comment:
This is not a service as it has a required composite parameter, thus another SPL app is needed to invoke it.

The parameter should default to a submission parameter, even that might have a default.